gene_finder.py

```diff
@@ -96,42 +122,68 @@ def find_all_ORFs(dna):
     >>> find_all_ORFs("ATGCATGAATGTAG")
     ['ATGCATGAATGTAG', 'ATGAATGTAG', 'ATG']
     """
-    # TODO: implement this
-    pass
+    a=[]     #New List
+    for i in range(3): #Look at all three frams, instead of one
+        a.extend(find_all_ORFs_oneframe(dna[i:])) #Add onto list a
+
+    # for i in range(0,len(dna),3):
+    #     if (dna[i:i+3]=='ATG'):
+    #         a.append(rest_of_ORF(dna[i:]))
+    #     elif (dna[i+1:i+4]=='ATG'):
+    #         a.append(rest_of_ORF(dna[i:]))
+    #     elif (dna[i+2:i+5]=='ATG'):
+    #         a.append(rest_of_ORF(dna[i:]))
+    return a   #Display results
 
 
 def find_all_ORFs_both_strands(dna):
     """ Finds all non-nested open reading frames in the given DNA sequence on both
         strands.
-
         dna: a DNA sequence
         returns: a list of non-nested ORFs
     >>> find_all_ORFs_both_strands("ATGCGAATGTAGCATCAAA")
     ['ATGCGAATG', 'ATGCTACATTCGCAT']
     """
-    # TODO: implement this
-    pass
-
+    a=[]                                        #New List
+    a.extend(find_all_ORFs(dna))                #add last functions results to new list
+    dna_reversed = dna[::-1]                    #Reverse direction of letters
+    d=""                                        #New String
+    for letter in dna_reversed:                 #look at each letter in new variable
+        reverse = get_complement(letter)        #get the complement of reversed letters
+        d+= reverse                             #add onto the string d
+    a.extend(find_all_ORFs(d))                  #add onto the list a
+    return a                                    #display results
 
 def longest_ORF(dna):
     """ Finds the longest ORF on both strands of the specified DNA and returns it
         as a string
     >>> longest_ORF("ATGCGAATGTAGCATCAAA")
     'ATGCTACATTCGCAT'
     """
-    # TODO: implement this
-    pass
-
+    count=0                                     #Start Count at 0
+    for i in find_all_ORFs_both_strands(dna):   #For items in the list created above
+        if len(i)>count:                        #Check their length to the last counted string
+            count = len (i)                     #If it is the longest, change it
+            a = i                               #make a the largest string
+    return a                                    #Display a
```

Comment:
just a note on naming variables. Naming it `max` or `longestORF` would be much more readable than naming it just `a`. 

Here's an article that's worth skimming through. http://archive.oreilly.com/pub/post/the_worlds_two_worst_variable.html